tomusborne/generatepress
# Fix: Kebab-case formatting for global colors variable names

This PR fixes and improves the kebab case transformation for global variables.

The bug originated from the Lodash Kebab function which splits the string into words and the numbers are considered single words causing a dash after/before a number. Introduced in the 3.2 version.

### How to reproduce
1. Go to Customizer > Colors
2. Add a new global color - name it: a1 and GP will make it a-1
3. Rewrite it as a1 and it should remain as that.
4. Then publish, exit the customizer and return and it will be a-1 again

The newly created function should transform
`---aA Bb c1 D2 e3F G4h #abc000 multi---------dashes thisIsMyWord !@#%^&*\(\)=-_ ---`
to
`a-a-bb-c1-d2-e3f-g4h-abc000-multi-dashes-this-is-my-word`.

Reference: https://generatepress.com/forums/topic/global-colors-has-unexpectedly-begun-adding-dashes-into-colour-names/?view=all#post-2427111
